src/eve/utils.py

```diff
@@ -555,37 +555,54 @@ def as_dict(self) -> Dict[str, T]:
     asdict = as_dict
 
 
-class FrozenNamespace(Namespace[T]):
-    """An immutable version of :class:`Namespace`.
+class ConstantNamespace(Namespace[T]):
+    """A version of :class:`Namespace` for collecting constant expressions.
 
     Examples:
-        >>> ns = FrozenNamespace(a=10, b="hello")
+        >>> ns = ConstantNamespace(a=10)
+        >>> ns.b = "hello"
         >>> ns.a
         10
         >>> ns.a = 20
         Traceback (most recent call last):
            ...
-        TypeError: Trying to modify immutable 'FrozenNamespace' instance.
+        TypeError: Trying to modify immutable member 'a' of a 'ConstantNamespace'.
 
-        >>> ns = FrozenNamespace(a=10, b="hello")
+        >>> ns = ConstantNamespace(a=10, b="hello")
         >>> list(ns.items())
         [('a', 10), ('b', 'hello')]
+    """
+
+    __slots__ = "__cached_hash_value__"  # This slot is used to avoid polluting the namespace
```

Comment:
This should also be moved out of ConstantNamespace (and optionally turned into a tuple)
```suggestion
    __slots__ = ("__cached_hash_value__",)  # This slot is used to avoid polluting the namespace
```

Reply:
Done. I also turned it into a tuple.